glib-macros/src/properties.rs

```diff
@@ -0,0 +1,631 @@
+            flags_paths,
+        }
+    }
+}
+
+fn expand_properties_fn(props: &[PropDesc]) -> TokenStream2 {
+    let n_props = props.len();
+    let crate_ident = crate_ident_new();
+    let properties_build_phase = props.iter().map(|prop| {
+        let PropDesc {
+            ty,
+            name,
+            nick,
+            blurb,
+            builder,
+            flags_paths,
+            ..
+        } = prop;
+
+        let flags = {
+            let write = prop.set.as_ref().map(|_| quote!(WRITABLE));
+            let read = prop.get.as_ref().map(|_| quote!(READABLE));
+
+            let flags_iter = [write, read].into_iter().flatten().chain(
+                prop.flags
+                    .iter()
+                    .map(|x| str::to_uppercase(x))
+                    .map(|f| TokenStream2::from_str(&f).unwrap()),
+            );
+            quote!(#crate_ident::ParamFlags::empty() #(| #crate_ident::ParamFlags::#flags_iter)* #(| #flags_paths)*)
+        };
+
+        let builder_call = builder
+            .as_ref()
+            .cloned()
+            .map(|(mut required_params, opts)| {
+                let name_expr = syn::ExprLit {
+                    attrs: vec![],
+                    lit: syn::Lit::Str(name.to_owned()),
+                };
+                required_params.insert(0, name_expr.into());
+                let required_params = required_params.iter();
+
+                quote!((#(#required_params,)*)#opts)
+            })
+            .unwrap_or(quote!((#name)));
+
+        let build_nick = nick.as_ref().map(|x| quote!(.nick(#x)));
+        let build_blurb = blurb.as_ref().map(|x| quote!(.blurb(#x)));
+        let span = prop.attrs_span;
+        quote_spanned! {span=>
+            <<#ty as #crate_ident::Property>::Value as #crate_ident::HasParamSpec>
+                ::param_spec_builder() #builder_call
+                #build_nick
+                #build_blurb
+                .flags(#flags)
+                .build()
+        }
+    });
+    quote!(
+        fn derived_properties() -> &'static [#crate_ident::ParamSpec] {
+            use #crate_ident::once_cell::sync::Lazy;
+            static PROPERTIES: Lazy<[#crate_ident::ParamSpec; #n_props]> = Lazy::new(|| [
+                #(#properties_build_phase,)*
+            ]);
+            PROPERTIES.as_ref()
+        }
+    )
+}
+fn expand_property_fn(props: &[PropDesc]) -> TokenStream2 {
+    let crate_ident = crate_ident_new();
+    let match_branch_get = props.iter().flat_map(|p| {
+        let PropDesc {
+            name,
+            field_ident,
+            member,
+            get,
+            ..
+        } = p;
+
+        let enum_ident = name_to_enum_ident(name.value());
+        let span = p.attrs_span;
+        get.as_ref().map(|get| {
+            let body = match (member, get) {
+                (_, MaybeCustomFn::Custom(expr)) => quote!(
+                    DerivedPropertiesEnum::#enum_ident => Ok((#expr)(&self).to_value())
```

Comment:
Now that #826 is merged, `to_value` can be replaced with `<#crate_ident::Value as ::std::convert::From>::from` here and anywhere else it is used

Reply:
@bilelmoussaoui this one is pending

Reply:
Did some changes but not  very sure about them

Reply:
Does it compile? Then it's most likely correct ;)